adal/src/main/java/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -169,7 +169,7 @@ public TokenCacheItem getItem(String key) {
             String json = mPrefs.getString(key);
             json = null != json ? json : "";
             String decrypted = decrypt(key, json);
-            if (decrypted != null) {
+            if (validTokenCacheItem(decrypted)) {
```

Comment:
Not just a logging enhancement.  You've added custom validation around whether this is JSON.  I think perhaps instead of validating yourself, we capture these additional hueristics after we receive the exception.

Reply:
Yes, after after throwing the error might be better. What seemed like is probably happening is the JSON is malformed, and so I wanted to know in which way it's malformed particularly. I did this to avoid logging the the tokenCacheItem itself since it'll validate PII. Do you think logging the first+last 5/10 chars would be safe?